src/main/java/seedu/address/logic/commands/EventCommand.java

```diff
@@ -20,28 +31,46 @@ public class EventCommand extends Command {
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an event to the address book.\n"
             + "Parameters: "
             + PREFIX_NAME + "EVENT NAME "
-            + PREFIX_DATE + "DATE (yyyy-mm-dd) \n"
+            + PREFIX_DATE + "DATE (yyyy-mm-dd) "
+            + PREFIX_ATTENDEES + "ATTENDEES BASED ON ADDRESS BOOK INDEXING \n"
             + "Example: "
             + COMMAND_WORD + " "
             + PREFIX_NAME + "New Year's Party "
-            + PREFIX_DATE + "2025-01-01";
+            + PREFIX_DATE + "2025-01-01 "
+            + PREFIX_ATTENDEES + "1 2 4 5";
 
     public static final String MESSAGE_SUCCESS = "New event added: %1$s";
     public static final String MESSAGE_DUPLICATE_EVENT = "This event already exists in the address book";
 
-    private final Event toAdd;
+    private final String eventName;
+    private final LocalDate eventDate;
+    private final Set<Index> attendeeIndexes;
 
     /**
      * Creates an EventCommand to add the specified {@code Event}
      */
-    public EventCommand(Event event) {
-        requireNonNull(event);
-        toAdd = event;
+    public EventCommand(String eventName, LocalDate eventDate, Set<Index> attendeeIndexes) {
+        requireAllNonNull(eventName, eventDate, attendeeIndexes);
+        this.eventName = eventName;
+        this.eventDate = eventDate;
+        this.attendeeIndexes = attendeeIndexes;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        Set<Person> attendees = new HashSet<>();
+        int size = model.getAddressBook().getPersonList().size();
+        for (Index index : this.attendeeIndexes) {
+            if (index.getZeroBased() >= size) {
+                throw new CommandException(MESSAGE_ATTENDEE_NOT_FOUND);
```

Comment:
Perhaps this error message should be `MESSAGE_INVALID_PERSON_DISPLAYED_INDEX`, what do you think?

Reply:
I think it's a little weird to put MESSAGE_INVALID_PERSON_DISPLAYED_INDEX because we aren't exactly displaying a person here. What do you think?

Reply:
I think we can leave as is.